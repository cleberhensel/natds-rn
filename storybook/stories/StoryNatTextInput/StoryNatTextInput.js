import React, { useState } from 'react';
import { Text, ScrollView, SafeAreaView } from 'react-native';
import propTypes from 'prop-types';
import { NatContainer, NatTextInput } from '../../../lib';

export const StoryNatTextInput = ({ type, icon }) => {
  const [inputSmall, setInputSmall] = useState('');
  const [inputNormal, setInputNormal] = useState('');

  return (
    <SafeAreaView>
      <NatContainer style={{ backgroundColor: '#FFF', width: 300, paddingTop: 20 }}>
        <ScrollView>
          <NatTextInput
            label="Size Small"
            placeholder="Size Small"
            assistiveText="Size Small"
            onChangeText={setInputSmall}
            value={inputSmall}
            size="small"
            type={type}
            icon={icon}
          />
          <NatTextInput
            label="Size Normal"
            placeholder="Size Normal"
            assistiveText="Size Normal"
            onChangeText={setInputNormal}
            value={inputNormal}
            size="normal"
            type={type}
            icon={icon}
          />
          <NatTextInput
            label="Multiple lines"
            placeholder="Multiple lines"
            assistiveText="Multiple lines"
            onChangeText={setInputNormal}
            value={inputNormal}
            size="normal"
            multiline
            numberOfLines={4}
            type={type}
            icon={icon}
          />
          <NatTextInput
            label=""
            placeholder="Search"
            onChangeText={setInputNormal}
            value={inputNormal}
            size="small"
            type={type}
            icon={icon}
          />
          <Text style={{ marginTop: 20 }}>
            You can apply margin-top by sending &apos;margin&apos; prop,
            with values: none, dense or normal.
          </Text>
        </ScrollView>
      </NatContainer>
    </SafeAreaView>
  );
};

StoryNatTextInput.propTypes = {
  type: propTypes.string.isRequired,
  icon: propTypes.isRequired,
};
