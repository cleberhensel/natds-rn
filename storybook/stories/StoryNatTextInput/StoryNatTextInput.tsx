import React, { useState } from 'react';
import { ScrollView, SafeAreaView, Dimensions } from 'react-native';
import { NatContainer, NatTextInput, NatText } from '../../../lib';

const styles = {
  defaultScreen: {
    flex: 1,
    width: Dimensions.get('window').width,
  },
};

type StoryNatTextInputProps = {
  icon?: string,
  type: string
};
export const StoryNatTextInput: React.SFC<StoryNatTextInputProps> = ({
  type,
  icon,
}) => {
  const [inputSmall, setInputSmall] = useState('');
  const [inputNormal, setInputNormal] = useState('');
  const [inputMultiline, setInputMultiline] = useState('');
  const [inputSearch, setInputSearch] = useState('');

  return (
    <SafeAreaView style={styles.defaultScreen}>
      <NatContainer>
        <ScrollView>
          <NatTextInput
            label="Size Small"
            placeholder="Size Small"
            assistiveText="Size Small"
            onChangeText={setInputSmall}
            value={inputSmall}
            size="small"
            type={type}
            icon={icon}
          />
          <NatTextInput
            label="Size Normal"
            placeholder="Size Normal"
            assistiveText="Size Normal"
            onChangeText={setInputNormal}
            value={inputNormal}
            size="normal"
            type={type}
            icon={icon}
          />
          <NatTextInput
            label="Multiple lines"
            placeholder="Multiple lines"
            assistiveText="Multiple lines"
            onChangeText={setInputMultiline}
            value={inputMultiline}
            size="normal"
            multiline
            numberOfLines={4}
            type={type}
            icon={icon}
          />
          <NatTextInput
            label=""
            placeholder="Search"
            onChangeText={setInputSearch}
            value={inputSearch}
            size="small"
            type={type}
            icon={icon}
          />
          <NatText
            value="body1. Lorem ipsum dolor sit amet, consectetur adipscing elit. Quos blanditiis tenetur unde suscipit, quam beatae rerum inventore consectetur, neque doloribus, cupiditate numquam dignissimos laborum fugiat deleniti? Eum quasi quidem quibusdam"
            type="body1"
          />
        </ScrollView>
      </NatContainer>
    </SafeAreaView>
  );
};
StoryNatTextInput.defaultProps = {
  icon: null,
};
