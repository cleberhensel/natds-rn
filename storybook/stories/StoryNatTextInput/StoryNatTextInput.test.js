import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components/native';

import { StoryNatTextInput } from './StoryNatTextInput';

describe('StoryNatTextInput component', () => {
  it('Should render StoryNatTextInput component ', () => {
    const button = renderer.create(
      <StoryNatTextInput />,
    ).toJSON();
    expect(button).toMatchSnapshot();
  });
});
