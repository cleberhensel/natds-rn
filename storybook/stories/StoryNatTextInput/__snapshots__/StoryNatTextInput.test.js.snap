// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoryNatTextInput component Should render StoryNatTextInput component  1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "bottom": 0,
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "left": 0,
        "paddingBottom": 0,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 0,
        "right": 0,
        "top": 0,
      },
      Object {
        "backgroundColor": "#FFF",
        "paddingTop": 20,
        "width": 300,
      },
    ]
  }
>
  <RCTScrollView>
    <View>
      <View
        accessible={true}
        defaultMargin="0px"
        style={
          Array [
            Object {
              "marginTop": 0,
            },
          ]
        }
      >
        <Text
          state="default"
          style={
            Array [
              Object {
                "color": "#777777",
                "fontSize": 14,
                "fontWeight": "500",
                "marginBottom": 3,
              },
            ]
          }
          type="default"
        >
          Size Small
        </Text>
        <View
          defaultSize="40px"
          state="default"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderColor": "#777777",
                "borderRadius": 4,
                "borderStyle": "solid",
                "borderWidth": 1,
                "flexDirection": "row",
                "height": 40,
                "justifyContent": "space-between",
                "paddingBottom": 0,
                "paddingLeft": 16,
              },
            ]
          }
          type="default"
        >
          <TextInput
            allowFontScaling={true}
            editable={true}
            multiline={false}
            numberOfLines={1}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholder="Size Small"
            rejectResponderTermination={true}
            state="default"
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "color": "#333333",
                  "flexBasis": 0,
                  "flexGrow": 7,
                  "flexShrink": 1,
                  "fontSize": 16,
                  "height": 56,
                  "letterSpacing": 0,
                },
                Object {
                  "height": 36,
                },
              ]
            }
            type="default"
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
        <Text
          state="default"
          style={
            Array [
              Object {
                "color": "#777777",
                "fontSize": 12,
                "fontWeight": "400",
                "marginTop": 3,
              },
            ]
          }
          type="default"
        >
          Size Small
        </Text>
      </View>
      <View
        accessible={true}
        defaultMargin="0px"
        style={
          Array [
            Object {
              "marginTop": 0,
            },
          ]
        }
      >
        <Text
          state="default"
          style={
            Array [
              Object {
                "color": "#777777",
                "fontSize": 14,
                "fontWeight": "500",
                "marginBottom": 3,
              },
            ]
          }
          type="default"
        >
          Size Normal
        </Text>
        <View
          defaultSize="56px"
          state="default"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderColor": "#777777",
                "borderRadius": 4,
                "borderStyle": "solid",
                "borderWidth": 1,
                "flexDirection": "row",
                "height": 56,
                "justifyContent": "space-between",
                "paddingBottom": 0,
                "paddingLeft": 16,
              },
            ]
          }
          type="default"
        >
          <TextInput
            allowFontScaling={true}
            editable={true}
            multiline={false}
            numberOfLines={1}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholder="Size Normal"
            rejectResponderTermination={true}
            state="default"
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "color": "#333333",
                  "flexBasis": 0,
                  "flexGrow": 7,
                  "flexShrink": 1,
                  "fontSize": 16,
                  "height": 56,
                  "letterSpacing": 0,
                },
                Object {
                  "height": 52,
                },
              ]
            }
            type="default"
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
        <Text
          state="default"
          style={
            Array [
              Object {
                "color": "#777777",
                "fontSize": 12,
                "fontWeight": "400",
                "marginTop": 3,
              },
            ]
          }
          type="default"
        >
          Size Normal
        </Text>
      </View>
      <View
        accessible={true}
        defaultMargin="0px"
        style={
          Array [
            Object {
              "marginTop": 0,
            },
          ]
        }
      >
        <Text
          state="default"
          style={
            Array [
              Object {
                "color": "#777777",
                "fontSize": 14,
                "fontWeight": "500",
                "marginBottom": 3,
              },
            ]
          }
          type="default"
        >
          Multiple lines
        </Text>
        <View
          defaultSize="160px"
          state="default"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderColor": "#777777",
                "borderRadius": 4,
                "borderStyle": "solid",
                "borderWidth": 1,
                "flexDirection": "row",
                "height": 160,
                "justifyContent": "space-between",
                "paddingBottom": 0,
                "paddingLeft": 16,
              },
            ]
          }
          type="default"
        >
          <TextInput
            allowFontScaling={true}
            editable={true}
            multiline={true}
            numberOfLines={4}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholder="Multiple lines"
            rejectResponderTermination={true}
            state="default"
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "color": "#333333",
                  "flexBasis": 0,
                  "flexGrow": 7,
                  "flexShrink": 1,
                  "fontSize": 16,
                  "height": 56,
                  "letterSpacing": 0,
                },
                Object {
                  "height": 156,
                },
              ]
            }
            type="default"
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
        <Text
          state="default"
          style={
            Array [
              Object {
                "color": "#777777",
                "fontSize": 12,
                "fontWeight": "400",
                "marginTop": 3,
              },
            ]
          }
          type="default"
        >
          Multiple lines
        </Text>
      </View>
      <View
        accessible={true}
        defaultMargin="0px"
        style={
          Array [
            Object {
              "marginTop": 0,
            },
          ]
        }
      >
        <Text
          state="default"
          style={
            Array [
              Object {
                "color": "#777777",
                "fontSize": 14,
                "fontWeight": "500",
                "marginBottom": 3,
              },
            ]
          }
          type="default"
        >
          
        </Text>
        <View
          defaultSize="40px"
          state="default"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderColor": "#777777",
                "borderRadius": 4,
                "borderStyle": "solid",
                "borderWidth": 1,
                "flexDirection": "row",
                "height": 40,
                "justifyContent": "space-between",
                "paddingBottom": 0,
                "paddingLeft": 16,
              },
            ]
          }
          type="default"
        >
          <TextInput
            allowFontScaling={true}
            editable={true}
            multiline={false}
            numberOfLines={1}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholder="Search"
            rejectResponderTermination={true}
            state="default"
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "color": "#333333",
                  "flexBasis": 0,
                  "flexGrow": 7,
                  "flexShrink": 1,
                  "fontSize": 16,
                  "height": 56,
                  "letterSpacing": 0,
                },
                Object {
                  "height": 36,
                },
              ]
            }
            type="default"
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
      <Text
        style={
          Object {
            "marginTop": 20,
          }
        }
      >
        You can apply margin-top by sending 'margin' prop, with values: none, dense or normal.
      </Text>
    </View>
  </RCTScrollView>
</View>
`;
